3.6.x regression `Provider io.micronaut.annotation.processing.TypeElementVisitorProcessor could not be instantiated`

```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] Bad service configuration file, or exception thrown while constructing Processor object: javax.annotation.processing.Processor: Provider io.micronaut.annotation.processing.TypeElementVisitorProcessor could not be instantiated: java.lang.ExceptionInInitializerError
```

To reproduce, checkout the `test-resources-integration` branch of the Maven plugin, then `mvn install` it, then try to compile the following application:

[demo.zip](https://github.com/micronaut-projects/micronaut-core/files/9065661/demo.zip)

Possibly caused by #7608 or #7651 

Blocks https://github.com/micronaut-projects/micronaut-maven-plugin/pull/398

--- a/inject-java/src/main/java/io/micronaut/annotation/processing/TypeElementVisitorProcessor.java
+++ b/inject-java/src/main/java/io/micronaut/annotation/processing/TypeElementVisitorProcessor.java
@@ -346,7 +346,7 @@ public class TypeElementVisitorProcessor extends AbstractInjectAnnotationProcess
             return true;
         }).stream()
             // remove duplicate classes
-            .collect(Collectors.toMap(v -> v.getClass(), v -> v)).values();
+            .collect(Collectors.toMap(v -> v.getClass(), v -> v, (a, b) -> a)).values();
     }
 
     /**

--- a/inject-java/src/test/resources/META-INF/micronaut/io.micronaut.inject.visitor.TypeElementVisitor/io.micronaut.visitors.InjectVisitor
+++ b/inject-java/src/test/resources/META-INF/micronaut/io.micronaut.inject.visitor.TypeElementVisitor/io.micronaut.visitors.InjectVisitor
@@ -1,0 +1,1 @@
+# duplicated in the normal service config to test deduplication in TypeElementVisitorProcessor

